servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultHttpServerBuilder.java

```diff
@@ -322,35 +308,59 @@ private HttpExecutionContext buildExecutionContext(final HttpExecutionStrategy s
            connectionAcceptorFactory.create(ACCEPT_ALL);
 
         final StreamingHttpService filteredService;
-        HttpExecutionContext serviceContext;
+        final HttpExecutionContext executionContext;
 
         if (noOffloadServiceFilters.isEmpty()) {
             filteredService = serviceFilters.isEmpty() ? rawService : buildService(serviceFilters.stream(), rawService);
-            serviceContext = buildExecutionContext(strategy);
+            executionContext = buildExecutionContext(strategy);
         } else {
             Stream<StreamingHttpServiceFilterFactory> nonOffloadingFilters = noOffloadServiceFilters.stream();
 
             if (strategy.hasOffloads()) {
-                serviceContext = buildExecutionContext(noOffloadsStrategy());
-                BooleanSupplier shouldOffload = serviceContext.ioExecutor().shouldOffloadSupplier();
+                executionContext = buildExecutionContext(noOffloadsStrategy());
+                BooleanSupplier shouldOffload = executionContext.ioExecutor().shouldOffloadSupplier();
                 // We are going to have to offload, even if just to the raw service
                 OffloadingFilter offloadingFilter =
                         new OffloadingFilter(strategy, buildFactory(serviceFilters), shouldOffload);
                 nonOffloadingFilters = Stream.concat(nonOffloadingFilters, Stream.of(offloadingFilter));
             } else {
                 // All the filters can be appended.
                 nonOffloadingFilters = Stream.concat(nonOffloadingFilters, serviceFilters.stream());
-                serviceContext = buildExecutionContext(strategy);
+                executionContext = buildExecutionContext(strategy);
             }
             filteredService = buildService(nonOffloadingFilters, rawService);
         }
 
-        return doListen(connectionAcceptor, serviceContext, filteredService, drainRequestPayloadBody)
+        if (noOffloadsStrategy() == strategy) {
```

Comment:
reminder: pending discussion in `DefaultSingleAddressHttpClientBuilder` we should be consistent here.